### PR TITLE
Improve warnings in UCR_UEA_datasets.load_dataset()

### PR DESCRIPTION
**Pull request by felixdivo**
While debugging #282, I improved the warnings that are raised by `tslearn.UCR_UEA_datasets.load_dataset()`. It now more clearly explains why things fail in the case of #282.

### CODE REVIEW

**Comment by felixdivo** [comment]
I've just seen that these changes are a bit dated an will rebase them next week.

**Comment by johannfaouzi** [comment]
In my project (https://github.com/johannfaouzi/pyts), I have a function to load datasets in `.arff` format. It uses [scipy.io.arff.loadarff](https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.arff.loadarff.html) and requires `scipy >= 1.3.0` to work with relational ARFF. Maybe worth adding to tslearn? 

[Relevant file](https://github.com/johannfaouzi/pyts/blob/master/pyts/datasets/uea.py), lines 269-273 and 286-314.

Basically, it loads the `.arff` files using `scipy.io.arff.loadarff` and then it transforms this object into a 3D numpy array, but the order of the dimension is different in my project. It could be easily adapted I think.

**Comment by felixdivo** [comment]
@johannfaouzi Thanks for the idea. I just realized that [on the current master branch of tslearn](https://github.com/tslearn-team/tslearn/blob/master/tslearn/datasets.py#L331). So #282 should actually work by now so I need to do some further digging. I might have a look at your implementation and will still port the warning logging over as I think that would be helpful if things go wrong in the future.
